public/scripts/last.js

```diff
@@ -104,9 +100,22 @@ function getFullHistory(requestLength, previousDate, todayDate) {
       type:'POST',
       url: 'http://ws.audioscrobbler.com/2.0/?method=user.getRecentTracks&api_key=100a45f60fce336c43b1dac55062e23a&username=' + userLastId + '&from='+ previousDate +'&to='+ todayDate +'&page='+ i +'&format=json',
       success: function(response) {
-        for (i = 0; i <= (response.recenttracks.track.length - 1); i ++){
-          allCallSongs.push(response.recenttracks.track[i]);
+
+        for (j = 0; j <= (response.recenttracks.track.length - 1); j ++){
+          allCallSongs.push(response.recenttracks.track[j]);
```

Comment:
note that even though we're in this inner anonymous `success` function, we still have access to the local variable `allCallSongs` from the containing `getFullHistory()` function. that's because javascript supports a language feature called "closures", which you may have heard of before. if you're not super familiar with what they are, look them up or ask me! closures are super cool. it's possible to go overboard with them, but if used judiciously they're a very useful tool.

Reply:
oh also there's no reason for me to have renamed `i` to `j` here, it would have been necessary in the first approach i was doing but i ended up doing something else, so feel free to change this back to `i`. my bad! this wasn't me trying to communicate anything about code style, it was just something i missed when cleaning up this diff for submission.